### PR TITLE
update Makefile target for cloudbuild-artifacts

### PR DESCRIPTION
**Pull request by prasita123**
Adds cloudbuild targets for ecr-credential-provider

**What type of PR is this?**
> Uncomment only one, leave it on its own line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by olemarkus** [comment]
Is it me, or is this the kops make targets?

**Comment by olemarkus** [comment]
Could you maybe fill in the PR template so we know what this is about?

File: Makefile
Comment (nckturner) [inline Makefile:199]: `nodeup` is a kops binary, this should be `ecr-credential-provider`, but we already have a build target for that, so what we need is a target to copy it. This is satisfied by `copy-bins-for-upload`.

**Review by nckturner — CHANGES_REQUESTED** [review]

File: Makefile
Comment (prasita123) [reply Makefile]: updated this part

File: Makefile
Comment (prasita123) [reply Makefile]: updated this part

**Comment by olemarkus** [comment]
Note that https://github.com/kubernetes/k8s.io/pull/4318 is needed before this PR can be completed.

**Comment by prasita123** [comment]
@olemarkus i have a similar pr https://github.com/kubernetes/k8s.io/pull/4067

File: cloudbuild.yaml
Comment (olemarkus) [inline cloudbuild.yaml:18]: This needs bumping now

File: cloudbuild.yaml
Comment (olemarkus) [inline cloudbuild.yaml:26]: The `$DOCKER*`  envs are probably not needed for this target

**Review by olemarkus — CHANGES_REQUESTED** [review]

**Comment by dims** [comment]
@prasita123 we landed https://github.com/kubernetes/k8s.io/pull/4318, we can restart the work on this PR.

File: Makefile
Comment (dims) [inline Makefile:31]: GCS location should be `gs://k8s-staging-provider-aws/releases/` before we land this PR, see [1]

[1] https://github.com/kubernetes/k8s.io/blob/main/artifacts/filestores/k8s-staging-provider-aws/filepromoter-manifest.yaml#L3

File: cloudbuild.yaml
Comment (nckturner) [inline cloudbuild.yaml:24]: So `$CI` is set by the prow job, but I'm not sure what is going on here.  Guessing this is an artifact from kops and can be removed?

Comment (olemarkus) [reply cloudbuild.yaml:24]: Yeah, this is a leftover from kOps. Think the makefile used this to determine how to create the build version. But the kOps Makefile does not contain any reference to this variable anymore. Can probably be removed.

File: Makefile
Comment (olemarkus) [reply Makefile:31]: The location should be set in the prow job, I think. The cloudbuild config below already forwards the env var.

File: cloudbuild.yaml
Comment (prasita123) [reply cloudbuild.yaml]: updated to 1.19.5

File: cloudbuild.yaml
Comment (prasita123) [reply cloudbuild.yaml]: removed references of this var

File: cloudbuild.yaml
Comment (prasita123) [reply cloudbuild.yaml]: removed reference to _CI

**Comment by dims** [comment]
/triage accepted
/approve
/lgtm

will leave it to @olemarkus to drop the hold

File: Makefile
Comment (dims) [reply Makefile]: it's ok to set it in both places @olemarkus

**Review by olemarkus — APPROVED** [review]
Think this should work. Let's see if we can get a staging artefact.

**Pull request merged**
